src/librustc/ty/query/caches.rs

```diff
@@ -23,7 +23,7 @@ pub(crate) trait QueryCache: Default {
     /// to compute it.
     fn lookup<'tcx, R, GetCache, OnHit, OnMiss>(
         &self,
-        state: &'tcx QueryState<'tcx, Self>,
+        lookup: QueryLookup<'tcx, Self::Key, Self::Sharded>,
```

Comment:
`QueryLookup` should not be passed here has `lookup` is designed to allow lock-free lookup of query caches while `QueryLookup` requires getting a lock. I'd just remove this commit.